src/main/java/com/auth0/AuthenticationController.java

```diff
@@ -202,34 +218,88 @@ public void doNotSendTelemetry() {
 
     /**
      * Processes a request validating the received parameters and performs a Code Exchange or a Token's Signature Verification,
-     * depending on the chosen Response Type, to finally obtain a set of {@link Tokens}.
+     * depending on the chosen Response Type, to obtain a set of {@link Tokens}.
+     *
+     * <p><strong>Important:</strong> When using this API, you <strong>must</strong> also use {@link AuthenticationController#buildAuthorizeUrl(HttpServletRequest, HttpServletResponse, String)}
+     * when building the {@link AuthorizeUrl} that the user will be redirected to to login. Failure to do so may result
+     * in a broken login experience for the user.</p>
+     *
+     * @param request the received request to process.
+     * @param response the received response to process.
+     * @return the Tokens obtained after the user authentication.
+     * @throws InvalidRequestException       if the error is result of making an invalid authentication request.
+     * @throws IdentityVerificationException if an error occurred while verifying the request tokens.
+     */
+    public Tokens handle(HttpServletRequest request, HttpServletResponse response) throws IdentityVerificationException {
+        Validate.notNull(request, "request must not be null");
+        Validate.notNull(response, "response must not be null");
+
+        return requestProcessor.process(request, response);
+    }
+
+    /**
+     * Processes a request validating the received parameters and performs a Code Exchange or a Token's Signature Verification,
```

Comment:
Agree. Think this was moved from an existing API, but will update for clarity in both places while working in this area.

Reply:
How about this?

> Process a request to obtain a set of {@link Tokens} that represent an authenticated user's credentials.
> 
> This method should be called when processing the callback request to your application. It will validate authentication-related request parameters, handle performing a Code Exchange request if using the "code" response type, and verify the integrity of the ID token (if present).

Reply:
Updated to apply suggestion.